plotly/matplotlylib/mpltools.py

```diff
@@ -453,13 +453,13 @@ def prep_ticks(ax, index, ax_type, props):
         else:
             axis_dict['tick0'] = tick0
             axis_dict['dtick'] = dtick
-            axis_dict['tickmode'] = False
+            axis_dict['tickmode'] = None
```

Comment:
I think it would be better to be explicit here and use `'auto'` rather than `None`

Reply:
Can you maybe help me to understand what is going on? Are you trying to catch the axis scaling properties {“linear”, “log”, “symlog”, “logit”} from [matplotlib.axes.Axes.set_xscale](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.set_xscale.html#matplotlib.axes.Axes.set_xscale) (other axis respectively)? There is no `auto` option. Maybe `None` is treated as auto?
It seems like despite the `tickmode` is set to `None`, it is inferred later on, for example the parsed JSON shows:
```...
'layout': {'autosize': False,
               'height': 360,
               'hovermode': 'closest',
               'margin': {'b': 41, 'l': 61, 'pad': 0, 'r': 79, 't': 14},
               'showlegend': False,
               'width': 434,
               'xaxis':{...
                         'type': 'linear',
                         ...},
               'yaxis': {'anchor': 'x',
                         ...
                         'type': 'linear',
                         ...}
...
```

Reply:
btw, i chose `None` in the first place because of the code snippet `axis_dict.pop('tickmode', None)` in line 491.